mm2src/lp_ordermatch.rs

```diff
@@ -2835,11 +2840,17 @@ pub async fn lp_ordermatch_loop(ctx: MmArc) {
             }
 
             for uuid in to_cancel {
-                if let Some(order) = my_maker_orders.remove(&uuid) {
-                    delete_my_maker_order(ctx.clone(), order, MakerOrderCancellationReason::InsufficientBalance)
-                        .compat()
-                        .await
-                        .ok();
+                let maybe_order_mutex = ordermatch_ctx.my_maker_orders.lock().remove(&uuid);
+                if let Some(order_mutex) = maybe_order_mutex {
+                    let order = order_mutex.lock().await;
```

Comment:
I agree with you, I see now that the orders were canceled in another loop because we were iterating over the hashmap we needed to remove the order from but now since we iterate over a clone of the map it should be done in the above loop.